mbedtls fails to build with ARC MWDT toolchain in case of minimal libc is used

**Describe the bug**

mbedtls fails to build with ARC MWDT toolchain in case of minimal libc is used. For example, the `sample.drivers.crypto.mbedtls`:

```
- [** snip **] -
[137/224] Building C object modules/mbedtls/CMakeFiles/mbedTLSBase.dir/verification_workspace/modules/crypto/mbedtls/library/padlock.c.obj
[138/224] Building C object modules/mbedtls/CMakeFiles/mbedTLSBase.dir/verification_workspace/modules/crypto/mbedtls/library/platform_util.c.obj
/verification_workspace/modules/crypto/mbedtls/library/platform_util.c:109:9: warning: call to undeclared function 'memset_s'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  109 |         memset_s(buf, len, 0, len);
      |         ^
1 warning generated.
[139/224] Building C object modules/mbedtls/CMakeFiles/mbedTLSBase.dir/verification_workspace/modules/crypto/mbedtls/library/platform.c.obj
[140/224] Building C object modules/mbedtls/CMakeFiles/mbedTLSBase.dir/verification_workspace/modules/crypto/mbedtls/library/poly1305.c.obj
[141/224] Building C object modules/mbedtls/CMakeFiles/mbedTLSBase.dir/verification_workspace/modules/crypto/mbedtls/library/psa_util.c.obj
   - [** snip **] -
[215/224] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timer.c.obj
[216/224] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mempool.c.obj
[217/224] Building C object zephyr/kernel/CMakeFiles/kernel.dir/dynamic_disabled.c.obj
[218/224] Linking C static library zephyr/kernel/libkernel.a
[219/224] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /verification_workspace/twister-out/nsim_nsim_hs/samples/drivers/crypto/sample.drivers.crypto.mbedtls/zephyr/zephyr_pre0.map 
: && ccache /mwdt/MetaWare/arc/bin/ccac   zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arc/core/offsets/offsets.c.obj  -Wl,-T/verification_workspace/twister-out/nsim_nsim_hs/samples/drivers/crypto/sample.drivers.crypto.mbedtls/zephyr/linker_zephyr_pre0.cmd  -Wl,--gc-sections  -Wl,--entry=__start  -Wl,--Map=/verification_workspace/twister-out/nsim_nsim_hs/samples/drivers/crypto/sample.drivers.crypto.mbedtls/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arc/core/libarch__arc__core.a  zephyr/arch/arch/arc/core/mpu/libarch__arc__core__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/crypto/libdrivers__crypto.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/mbedtls/libmbedTLSBase.a  modules/mbedtls/libmbedTLSCrypto.a  modules/mbedtls/libmbedTLSX509.a  modules/mbedtls/libmodules__mbedtls.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L/verification_workspace/twister-out/nsim_nsim_hs/samples/drivers/crypto/sample.drivers.crypto.mbedtls/zephyr  zephyr/arch/common/libisr_tables.a  modules/mbedtls/libmbedTLSBase.a  modules/mbedtls/libmbedTLSCrypto.a  modules/mbedtls/libmbedTLSX509.a  -Wl,--sort-section=alignment  -Wl,-u_OffsetAbsSyms  -Wl,-u_ConfigAbsSyms  -Hlld  -Hnosdata  -Xtimer0  -Hnoxcheck_obj  -Hnocplus  -Hhostlib=  -Hheap=0  -Hnoivt  -Hnocrt  -Hnolib  -Hldopt=-lmw  -Hlib=hs38_full && cd /verification_workspace/twister-out/nsim_nsim_hs/samples/drivers/crypto/sample.drivers.crypto.mbedtls/zephyr && /usr/bin/cmake -E true
LLD 17.0.7
lldac: error: undefined symbol: memset_s
>>> referenced by platform_util.c:109 (/verification_workspace/modules/crypto/mbedtls/library/platform_util.c:109)
>>>               platform_util.c.obj:(mbedtls_platform_zeroize) in archive modules/mbedtls/libmbedTLSBase.a
ninja: build stopped: subcommand failed.
```

**Expected behavior**
No build error should happen

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: MWDT 2024.06
 - Commit SHA: recent main, 1ed04e7b49c5bec8e09f81480b7f95ae51b59b10

--- a/cmake/compiler/arcmwdt/compiler_flags.cmake
+++ b/cmake/compiler/arcmwdt/compiler_flags.cmake
@@ -116,7 +116,7 @@ set_compiler_property(PROPERTY warning_error_misra_sane -Werror=vla)
 set_compiler_property(PROPERTY cstd -std=)
 
 if (NOT CONFIG_ARCMWDT_LIBC)
-  set_compiler_property(PROPERTY nostdinc -Hno_default_include -Hnoarcexlib)
+  set_compiler_property(PROPERTY nostdinc -Hno_default_include -Hnoarcexlib -U__STDC_LIB_EXT1__)
   set_compiler_property(APPEND PROPERTY nostdinc_include ${NOSTDINC})
 endif()
 

